src/gui/widgets/general_settings_page.cpp

```diff
@@ -173,17 +174,43 @@ void GeneralSettingsPage::apply()
 		case QLocale::AnyLanguage:
 		case QLocale::C:
 		case QLocale::English:
-			QMessageBox::information(window(), QLatin1String("Notice"), QLatin1String("The program must be restarted for the language change to take effect!"));
-			break;
+			{
+				QMessageBox messageBox(window());
+				messageBox.setWindowTitle(QLatin1String("Restart?"));
+				messageBox.setText(QLatin1String("The program must be restarted for the language change to take effect!"));
+				messageBox.setInformativeText(QLatin1String("Do you want to restart now?"));
+				messageBox.setIcon(QMessageBox::Question);
+				auto yesButton = messageBox.addButton(QLatin1String("Restart Now"), QMessageBox::YesRole);
+				messageBox.addButton(QLatin1String("Restart Later"), QMessageBox::NoRole);
+				messageBox.exec();
+				if (messageBox.clickedButton() == yesButton)
+				{
+					qApp->exit(MainWindow::EXIT_CODE_REBOOT);
+				}
+				break;
+			}
 			
 		default:
-			qApp->installEventFilter(this);
-			qApp->installTranslator(&translation.getQtTranslator());
-			qApp->installTranslator(&translation.getAppTranslator());
-			QMessageBox::information(window(), tr("Notice"), tr("The program must be restarted for the language change to take effect!"));
-			qApp->removeTranslator(&translation.getAppTranslator());
-			qApp->removeTranslator(&translation.getQtTranslator());
-			qApp->removeEventFilter(this);
+			{
+				qApp->installEventFilter(this);
+				qApp->installTranslator(&translation.getQtTranslator());
+				qApp->installTranslator(&translation.getAppTranslator());
+				QMessageBox messageBox(window());
+				messageBox.setWindowTitle(tr("Restart?"));
+				messageBox.setText(tr("The program must be restarted for the language change to take effect!"));
+				messageBox.setInformativeText(tr("Do you want to restart now?"));
+				messageBox.setIcon(QMessageBox::Question);
+				auto yesButton = messageBox.addButton(tr("Restart Now"), QMessageBox::YesRole);
+				messageBox.addButton(tr("Restart Later"), QMessageBox::NoRole);
+				messageBox.exec();
+				if (messageBox.clickedButton() == yesButton)
+				{
+					qApp->exit(MainWindow::EXIT_CODE_REBOOT);
+				}
```

Comment:
I would suggest to deduplicate the message box part. This requires to solve the different need for tr(). Maybe the easiest way is to put this in a member function which takes a function pointer named tr, receiving either a pointer to tr(), or a no-op lambda.